src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ChunkInlineOutput.java

```diff
@@ -0,0 +1,113 @@
+
+/*
+ * ChunkInlineOutput.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.studio.client.workbench.views.source.editors.text;
+
+import org.rstudio.core.client.VirtualConsole;
+import org.rstudio.core.client.widget.MiniPopupPanel;
+import org.rstudio.core.client.widget.PreWidget;
+import org.rstudio.studio.client.workbench.views.console.events.ConsoleWriteErrorEvent;
+import org.rstudio.studio.client.workbench.views.console.events.ConsoleWriteErrorHandler;
+import org.rstudio.studio.client.workbench.views.console.events.ConsoleWriteOutputEvent;
+import org.rstudio.studio.client.workbench.views.console.events.ConsoleWriteOutputHandler;
+import org.rstudio.studio.client.workbench.views.source.editors.text.ace.Range;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.resources.client.ClientBundle;
+import com.google.gwt.resources.client.CssResource;
+
+public class ChunkInlineOutput extends MiniPopupPanel
+                               implements ConsoleWriteOutputHandler,
+                                          ConsoleWriteErrorHandler
+{
+   public enum State
+   {
+      Queued,
+      Started,
+      Finished
+   }
+
+   public ChunkInlineOutput(String chunkId, Range range) 
+   {
+      super(true, false, true);
+      
+      vconsole_ = new VirtualConsole();
+      console_ = new PreWidget();
+      chunkId_ = chunkId;
+      range_ = range;
```

Comment:
I think these need to be anchored ranges -- otherwise, as the associated line is moved around, we'll lose track of it and we'll end up generating duplicate popups when re-executing that inline chunk.